compiler-plugin/src/main/java/io/ballerina/stdlib/ftp/plugin/FtpFunctionValidator.java

```diff
@@ -85,16 +89,105 @@ private void validateFunctionParameters(SeparatedNodeList<ParameterNode> paramet
         if (parameters.size() == 1) {
             ParameterNode paramNode = parameters.get(0);
             SyntaxKind paramSyntaxKind = ((RequiredParameterNode) paramNode).typeName().kind();
-            if (paramSyntaxKind.equals(SyntaxKind.QUALIFIED_NAME_REFERENCE)) {
+            if (paramSyntaxKind.equals(QUALIFIED_NAME_REFERENCE)) {
                 validateWatchEventParam(paramNode);
-            } else if (paramSyntaxKind.equals(SyntaxKind.INTERSECTION_TYPE_DESC)) {
-                validateIntersectionParam(paramNode);
+            } else if (paramSyntaxKind.equals(INTERSECTION_TYPE_DESC)) {
+                boolean watchEventParam = validateIntersectionParam(paramNode);
+                if (!watchEventParam) {
+                    context.reportDiagnostic(PluginUtils.getDiagnostic(
+                            INVALID_WATCHEVENT_PARAMETER,
+                            DiagnosticSeverity.ERROR, paramNode.location()));
+                }
             } else {
                 context.reportDiagnostic(PluginUtils.getDiagnostic(
-                        INVALID_PARAMETER,
+                        INVALID_WATCHEVENT_PARAMETER,
                         DiagnosticSeverity.ERROR, paramNode.location()));
             }
-        } else if (parameters.size() > 1) {
+        } else if (parameters.size() == 2) {
```

Comment:
I and @dilanSachi had a discussion decided to refactor the code a bit.

Reply:
Updated in https://github.com/ballerina-platform/module-ballerina-ftp/pull/580/commits/a277372468235ce743682f82bfe87cd00c6ea04f and https://github.com/ballerina-platform/module-ballerina-ftp/pull/580/commits/057084652d08880057e013e8d828ef2d5ea5b294